Sources/Public/iOS/AnimatedSwitch.swift

```diff
@@ -160,11 +163,19 @@ open class AnimatedSwitch: AnimatedControl {
       hapticGenerator.generateImpact()
     }
 
-    animationView.play(fromProgress: startProgress, toProgress: endProgress, loopMode: LottieLoopMode.playOnce) { finished in
-      if finished == true {
-        self.animationView.currentProgress = finalProgress
-      }
-    }
+    animationView.play(
+      fromProgress: startProgress,
+      toProgress: endProgress,
+      loopMode: LottieLoopMode.playOnce,
+      completion: { [weak self] finished in
+        guard let self = self else { return }
+
+        // For the legacy rendering engine, we freeze the animation at the expected final progress
+        // once the animation is complete. This isn't necessary on the new / experimental engine.
+        if finished, !(self.animationView.animationLayer is ExperimentalAnimationLayer) {
```

Comment:
#1478 also broke `AnimatedSwitch` animations somehow, not really sure why. This change fixes it.

![2022-01-13 09 42 26](https://user-images.githubusercontent.com/1811727/149381742-3aa17223-292c-4a50-8739-ac5618d13402.gif)


We don't currently have automated tests for `AnimatedSwitch`. I spent some time this morning trying to set up an automated test that called this, waited for the animation to finish, and then validated that the final `currentProgress` was correct. I wasn't able to get the animation to play correctly in a headless environment (the completion handler was always called immediately with `finished: false`).

Reply:
😞 bummer, glad the new engine doesn't need this though